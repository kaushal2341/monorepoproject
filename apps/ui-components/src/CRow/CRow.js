import React from 'react';
import Row from "react-bootstrap/Row";
import PropTypes from 'prop-types';

const CRow = props => {
    const {
        as,
        bsPrefix,
        children,
        className,
        id,
        noGutters
    } = props;
    return (
        <Row
            as={as}
            bsPrefix={bsPrefix}
            children={children}
            className={className}
            id={"row_".concat(id)}
            noGutters={noGutters}
        />
    )
};

CRow.propTypes = {
    as: PropTypes.elementType,
    bsPrefix: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    id: PropTypes.string.isRequired,
    noGutters: PropTypes.bool
};

export default CRow;