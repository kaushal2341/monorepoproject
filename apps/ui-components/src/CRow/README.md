# CRow 

Row component made using react-bootstrap's `Row` component.
To use it ,first import the component,

```jsx harmony
import {CRow} from '@cogent/ui-components';
```
Then use it as,

```jsx harmony
<CRow id="passwordRow">
      <CCol id="passLabel">
            <CFLabel
                 id="password"
                 labelName="Password"
                 column={true}
                 srOnly={false}
                 md={4}
            />
      </CCol>
      <CCol id="passwordCol">
           <CFControl
                id="password"
                type="password"
                placeholder="Your password here."
                size="lg"
                autoComplete="off"
           />
      </CCol>
</CRow>
```
## CRow Props

```
as: elementType - You can use a custom element type for this component.

bsPrefix string - defaultValue: 'row' - Change the underlying component CSS base class name and modifier class names prefix. This is an escape hatch for working with heavily customized bootstrap css.

children: node - Children components.

className: string - Css classname for additional styling.

id: string - Id of the component.

noGutters: boolean - defaultValue: false - 	
           Removes the gutter spacing between Cols as well as any added negative margins.
```