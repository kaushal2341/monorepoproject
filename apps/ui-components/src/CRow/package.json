{
  "name": "@cogent/c-row",
  "version": "0.0.0",
  "main": "CRow.js",
  "author": "Sabu Shakya",
  "license": "MIT",
  "private": true
}
