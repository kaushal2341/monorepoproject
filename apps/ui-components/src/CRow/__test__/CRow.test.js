import CRow from "../CRow";

expect.addSnapshotSerializer(enzymeSerializer);

describe('CRow Component Tests', () => {

    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<CRow id="test"/>);
    });

    it('should be defined', () => {
        expect(wrapper.find('CRow')).toBeDefined();
    });

    it('should have only one Form.Row component', () => {
        expect(wrapper.find('#row_test').length).toBe(1);
    });

    it('should have all props available', () => {
        let propsAvailable = [
            'as',
            'bsPrefix',
            'children',
            'className',
            'id',
            'noGutters'
        ];
        let propsOfCrow = Object.keys(wrapper.find('#row_test').props());
        propsAvailable.forEach((propAvail, i) => (
            expect(propAvail).toContain(propsOfCrow[i])
        ));
    });

    describe('Snapshot Testing', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        })
    });
});