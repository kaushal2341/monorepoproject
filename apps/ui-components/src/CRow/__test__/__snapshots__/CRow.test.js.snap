// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CRow Component Tests Snapshot Testing should match snapshot 1`] = `
<ForwardRef
  id="row_test"
  noGutters={false}
/>
`;
